eng/common/scripts/modules/Package-Properties.psm1

```diff
@@ -0,0 +1,252 @@
+        $pkgVersion = $projectData.project.version
+
+        if ($projectPkgName -eq $pkgName)
+        {
+            return [PackageProps]::new($pkgName, $pkgVersion.ToString(), $pkgPath)
+        }
+    }
+    return $null
+}
+
+# Takes package name and service Name
+# Returns important properties of the package as related to the language repo
+# Returns a PS Object with properties @ { pkgName, pkgVersion, pkgDirectoryPath, pkgReadMePath, pkgChangeLogPath }
+# Note: python is required for parsing python package properties.
+function Get-PkgProperties
+{
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [string]$PackageName,
+        [Parameter(Mandatory=$true)]
+        [string]$ServiceName,
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("net","java","js","python")]
+        [string]$Language,
+        [string]$RepoRoot="${PSScriptRoot}/../../../.."
+    )
+
+    $pkgDirectoryName = $null
+    $pkgDirectoryPath = $null
+    $serviceDirectoryPath = Join-Path $RepoRoot "sdk" $ServiceName
+    if (!(Test-Path $serviceDirectoryPath))
+    {
+        Write-Error "Service Directory $ServiceName does not exist"
+        exit 1
+    }
+
+    $directoriesPresent = Get-ChildItem $serviceDirectoryPath -Directory
+
+    foreach ($directory in $directoriesPresent)
+    {
+        $pkgDirectoryPath = Join-Path $serviceDirectoryPath $directory.Name
+        $pkgProps = Extract-PkgProps -pkgPath $pkgDirectoryPath -pkgName $PackageName -lang $Language
+        if ($pkgProps -ne $null)
+        {
+            return $pkgProps
+        }
+    }
+    Write-Error "Failed to retrive Properties for $PackageName"
+}
+
+# Takes ServiceName, Language, and Repo Root Directory
+# Returns important properties for each package in the specified service, or entire repo if the serviceName is not specified
+# Returns an Table of service key to array values of PS Object with properties @ { pkgName, pkgVersion, pkgDirectoryPath, pkgReadMePath, pkgChangeLogPath }
+function Get-AllPkgProperties
+{
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("net","java","js","python")]
+        [string]$Language,
+        [string]$RepoRoot="${PSScriptRoot}/../../../..",
+        [string]$ServiceName=$null
+    )
+
+    $pkgPropsResult = @{}
+
+    if ([string]::IsNullOrEmpty($ServiceName))
+    {
+        $searchDir = Join-Path $RepoRoot "sdk"
+        foreach ($dir in (Get-ChildItem $searchDir -Directory))
+        {
+            $serviceDir = Join-Path $searchDir $dir.Name
+
+            if (Test-Path (Join-Path $serviceDir "ci.yml"))
+            {
+                $activePkgList = Get-PkgListFromYml -ciYmlPath (Join-Path $serviceDir "ci.yml")
+                if ($activePkgList -ne $null)
+                {
+                    $pkgPropsforService = Operate-OnPackages -activePkgList $activePkgList -serviceName $dir.Name -language $Language -repoRoot $RepoRoot
+                    $pkgPropsResult[$dir.Name] = $pkgPropsforService
```

Comment:
Why are we putting these in a dictionary? I suspect it might be better to just be a list. I do think it might be worth adding the service name as a property in the PackageProps class.

Reply:
The properties are captured in a service basis. So for listing all package properties i need to still capture it. At best it will be a list containing Lists.

Reply:
I guess I can also just accumulate it into a single list.

Reply:
Yes I think accumulating them in a single list makes sense and if it is easy it might be worth passing the service directory down so it can end up in the package properties data.

Reply:
Fixed